packages/event-handler/src/rest/utils.ts

```diff
@@ -401,22 +426,21 @@ const streamifyResponse =
     return (async (event, responseStream, context) => {
       await handler(event, responseStream, context);
 
-      /* v8 ignore else -- @preserve */
-      if ('chunks' in responseStream && Array.isArray(responseStream.chunks)) {
-        const output = Buffer.concat(responseStream.chunks as Buffer[]);
-        const nullBytes = Buffer.from([0, 0, 0, 0, 0, 0, 0, 0]);
-        const separatorIndex = output.indexOf(nullBytes);
-
-        const preludeBuffer = output.subarray(0, separatorIndex);
-        const bodyBuffer = output.subarray(separatorIndex + 8);
-        const prelude = JSON.parse(preludeBuffer.toString());
-
-        return {
-          body: bodyBuffer.toString(),
-          headers: prelude.headers,
-          statusCode: prelude.statusCode,
-        } as TResult;
-      }
+      /* v8 ignore next -- @preserve */
+      const output: Buffer =
+        (responseStream as IResponseStream).getBuffer?.() ?? Buffer.from([]);
```

Comment:
This cast is necessary because of the mismatch between the types in the  `aws-lambda`.types package.